python/ray/serve/_private/deployment_state.py

```diff
@@ -1369,34 +1410,39 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
-
-        app_msg = f" in application '{self.app_name}'" if self.app_name else ""
-        logger.info(
-            f"Deploying new version of deployment {self.deployment_name}{app_msg}."
+        # Determine whether the updated target state is simply scaling
+        is_scale = target_state.equal_to_state_excluding_num_replicas(
+            self._target_state
         )
+        # If deployment is not in the middle of updating, and this new
+        # target state only changes the number of replicas, then set
+        # status to be UPSCALING/DOWNSCALING
+        if self._target_state.info and is_scale:
```

Comment:
Why would we mark state as updating if _target_state.info is absent?

Reply:
If it is absent, then this is the first time `deploy()` has been called i.e. it's a new deployment.

Reply:
Thanks for clarifying! Would be great to call out this context in the comment for future readers